cartography/intel/azure/sql.py

```diff
@@ -0,0 +1,855 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+
+import neo4j
+from azure.core.exceptions import HttpResponseError
+from azure.mgmt.sql import SqlManagementClient
+from msrestazure.azure_exceptions import CloudError
+
+from .util.credentials import Credentials
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_client(credentials: Credentials, subscription_id: str) -> SqlManagementClient:
+    """
+    Getting the Azure SQL client
+    """
+    client = SqlManagementClient(credentials, subscription_id)
+    return client
+
+
+@timeit
+def get_server_list(credentials: Credentials, subscription_id: str) -> List[Dict]:
+    """
+    Returning the list of Azure SQL servers.
+    """
+    try:
+        client = get_client(credentials, subscription_id)
+        server_list = list(map(lambda x: x.as_dict(), client.servers.list()))
+
+    except HttpResponseError as e:
```

Comment:
Non blocker: Can you try and get specific on some of the response error reasons? example:
https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L15

Reply:
Sure! 👍 I'll commit this change and ping you once it is done. I could get specific on ResourceNotFoundError and ClientAuthenticationError under HttpResponseError.

Reply:
@ramonpetgrave64 I have made the changes in the recent commit :D

Reply:
Alright. In a future CR, let's see if we can implement this exception handling into a single function, so we don't repeat code.

Reply:
Alright! I thought of doing that but making it as a single function would have generalized the exception handling. Like if we get a ResourceNotFoundError, the function would just print "Azure resource not found" which would hardly be helpful as we wouldn't know which resource was not found and for what resource the exception was raised. This approach kinda personalizes the error message that will be displayed.